src/util/scoped_timer.cpp

```diff
@@ -34,25 +35,27 @@ struct state {
     std::timed_mutex m_mutex;
     unsigned ms { 0 };
     event_handler * eh { nullptr };
-    int work { 0 };
+    bool working { false };
     std::condition_variable_any cv;
 };
 
-/*
- * NOTE: this implementation deliberately leaks threads when Z3
- * exits. this is preferable to deallocating on exit, because
- * destructing threads blocked on condition variables leads to
- * deadlock.
- */
-static std::vector<state*> available_workers;
+static ptr_vector<state> available_workers;
```

Comment:
vector::clear will not call "delete", so finalize will not really clean up memory. This is easy for me or Nuno to fix.

Based on Nuno's input I would say we don't have the finalize_scoped_timer registered as a finalizer, but we call it explicitly in api_context.cpp to clean up state for DLLs. This is the only scenario that really matters (and the command-line versions to quiet down the dynamic tools that would start complaining).